_docs/for-contributors/2017-01-02-implementation.md

```diff
@@ -59,25 +59,32 @@ block (if you're the selected stakeholder), repeat.
 
 ### Listeners
 
-Listeners handle incoming messages and respond to them. Various
-supplemental listeners will not be covered, focusing on the main ones
-instead:
+Listeners handle incoming messages and respond to them. 
+Various supplemental listeners will not be covered, focusing on the main ones instead.
+
+Mostly listeners use *Relay* framework, which inludes three type of messages:
```

Comment:
Maybe make sense to add link to high level description of what is *Relay*: _Inv/Req/Data and MessagePart_ in https://cardano-docs.iohk.io/protocols/csl-application-level/